poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -1185,4 +1240,260 @@ public void syncPDUID(FacilitySoftwareConfig powerIQ) {
          offset += limit;
       }
    }
+
+   public Map<String,Asset> getPDUIDAndAssetMap(List<Asset> pdusFromFlowgate){
+      Map<String,Asset> pudIdAndAssetMap = new HashMap<String,Asset>();
+      if(pdusFromFlowgate.isEmpty()) {
+         return pudIdAndAssetMap;
+      }
+      for(Asset pdu : pdusFromFlowgate) {
+         HashMap<String,String> justficationFields = pdu.getJustificationfields();
+         String pduInfo = justficationFields.get(FlowgateConstant.PDU);
+         if(pduInfo != null) {
+            Map<String,String> pduInfoMap = null;
+            try {
+               pduInfoMap = getPduInfoMap(pduInfo);
+            } catch (IOException e) {
+               logger.error("Sync PDU from PowerIQ error",e.getCause());
+               continue;
+            }
+            pudIdAndAssetMap.put(pduInfoMap.get(FlowgateConstant.PDU_ID_FROM_POWERIQ), pdu);
+         }
+      }
+      return pudIdAndAssetMap;
+   }
+
+   private String generatePduOutletString(List<Outlet> outletsFromPowerIQ) throws JsonProcessingException {
```

Comment:
the parameter name would be better to change to pduOutlets  since this method just try to serialize the outlets data.